qiskit/_schema_validation.py

```diff
@@ -0,0 +1,249 @@
+                       in `_SCHEMAS`.
+        check_schema (bool): Verify schema is valid.
+
+        validator_class (jsonschema.IValidator): JsonSchema IValidator
+                instance. Default behavior is to determine this from the schema
+                `$schema` field.
+
+        **validator_kwargs: Additional keyword arguments for validator.
+
+    Return:
+        jsonschema.IValidator: Validator for JSON schema.
+
+    Raises:
+        SchemaValidationError: Raised if validation fails.
+    """
+    if schema is None:
+        try:
+            schema = _SCHEMAS[name]
+        except KeyError:
+            raise SchemaValidationError('''Valid schema name or schema must
+                                           be provided.''')
+
+    if name not in _VALIDATORS:
+
+        if not isinstance(schema, dict):
+            raise SchemaValidationError('''Supplied schema must be a Python
+                                        dictionary with proper schema form.''')
+
+        # Resolve Json spec from schema if needed
+        if validator_class is None:
+            validator_class = jsonschema.validators.validator_for(schema)
+
+        # Generate and store validator in _VALIDATORS
+        _VALIDATORS[name] = validator_class(schema, **validator_kwargs)
+
+    validator = _VALIDATORS[name]
+
+    if check_schema:
+        validator.check_schema(schema)
+
+    return validator
+
+
+def _load_schemas_and_validators():
+    """
+    Load all default schemas into `_SCHEMAS`
+    """
+
+    schema_base_path = qiskit_path[0]
+    for name, path in _DEFAULT_SCHEMA_PATHS.items():
+        _load_schema(os.path.join(schema_base_path, path), name)
+        _create_validator(name)
+
+
+# Load all schemas on import
+_load_schemas_and_validators()
+
+
+def validate_json_against_schema(json_dict, schema,
+                                 err_msg=None):
+    """
+    Validates JSON dict against a schema.
+
+    Args:
+        json_dict (dict): Json to be validated.
+        schema (dict or str): Json schema or string of name in _VALIDATORS. If
+                              schema is provided `jsonschema.validate` will be
+                              called. If schema name is provided a validator
+                              will be created or recovered from
+                              `_VALIDATORS` cache.
+        err_msg (str): Optional error message.
+
+    Raises:
+        SchemaValidationError: Raised if validation fails.
+
+    """
+
+    try:
+        if isinstance(schema, str):
+            schema_name = schema
+            schema = _SCHEMAS[schema]
+            validator = _create_validator(schema_name)
+            validator.validate(json_dict)
+        else:
+            jsonschema.validate(json_dict, schema)
+    except jsonschema.ValidationError as err:
+        if err_msg is None:
+            err_msg = '''JSON failed validation. Set Qiskit log level to DEBUG
```

Comment:
Don't use multiline strings for the cases where your string does not fit in a line. This has the annoying effect of adding the break line and spaces you used for indenting the code so:

```python
print('''Hello
      world''')
```
will print:
```
Hello
      world
```
Instead, use [several strings in different lines](https://docs.python.org/3.7/reference/lexical_analysis.html#string-literal-concatenation) if you are inside a [parenthesis/bracket context](https://docs.python.org/3.7/reference/lexical_analysis.html#implicit-line-joining):
```python
print('Hello '
      'world')
```
will print:
```
Hello world
```
You can also use the scaping character `\` to change the line when not in a bracket context:
```python
err_msg = 'Hello ' \
          'world'
print(err_msg)
```
will print
```
Hello world
```
Please, remember to address all the apparitions of this issue.